src/main/python/jobs.py

```diff
@@ -226,455 +93,209 @@ def on_submit_clicked(self):
-        self.expect_time.setObjectName("Page_input_input")
-        line_layout.addWidget(box)
+        class Res(Enum):
+            EMPTY_ERROR = "Please enter value of MEMORY."
+            INT_ERROR = "Please enter an interger input"
+            SUCCESS = auto()
 
-        self.submission_hint = add_label(line_frame, "", name="Page_hint", align=Qt.AlignVCenter)
-        line_layout.addWidget(self.submission_hint)
+        # check if input is acceptable
+        res = job_input_check(self.memory.text(), Res)
 
-        spacer = QSpacerItem(0, 0, QSizePolicy.Expanding, QSizePolicy.Minimum)
-        line_layout.addItem(spacer)
+        if res is not Res.SUCCESS:
+            self.submission_hint.setText(res.value)
 
-        self.submit_button = add_button(line_frame, "SUBMIT", name="Page_button")
-        line_layout.addWidget(self.submit_button)
+            return False
 
-        # --------- end job_submission, spacer ------------
+        return True
 
-        spacer = QSpacerItem(0, 0, QSizePolicy.Minimum, QSizePolicy.Expanding)
-        window_layout.addItem(spacer)
+    def _on_source_check(self):
 
-    def enable_scheme_01_frame(self, event):
-        set_frame(self, self.scheme_01_frame)
-        self.setStyleSheet(self.cxt.jobs_style)
-        self.select_scheme = 1
+        # clean up hint
+        self.submission_hint.reset()
 
-    def enable_scheme_02_frame(self, event):
-        set_frame(self, self.scheme_02_frame)
-        self.setStyleSheet(self.cxt.jobs_style)
-        self.select_scheme = 2
-        
-    def enable_scheme_03_frame(self, event):
-        set_frame(self, self.scheme_03_frame)
-        self.setStyleSheet(self.cxt.jobs_style)
-        self.select_scheme = 3
+        if self.source_file.text() is "":
+            self.submission_hint.setText("Please enter a source file.")
+            return False
+        return True
 
-    def enable_scheme_04_frame(self, event):
-        set_frame(self, self.scheme_04_frame)
-        self.setStyleSheet(self.cxt.jobs_style)
-        self.select_scheme = 4
+    def _on_input_check(self):
 
+        # clean up hint
+        self.submission_hint.reset()
 
-class JobList(QFrame):
+        if self.input_file.text() is "":
+            self.submission_hint.setText("Please enter a input file.")
+            return False
+        return True
 
-    def __init__(self, cxt:ApplicationContext, *args, **kwargs):
-        super(QFrame, self).__init__(*args, **kwargs)
+    def _api_post_call(self, endpoint: str, dat: dict):
+        with Api(endpoint) as api:
+            status, res = api.post(dat)
 
-        # variable
-        self.table = None                   # widget
-        self.search_bar = None              # input widget
-        self.refresh_button = None          # button widget
-        self.remove_button = None           # button widget
+            if not res:
```

Comment:
This can be shortened to: `return res and status == 200`

Reply:
fixed